index.mjs

```diff
@@ -0,0 +1,76 @@
+//. # Fluture Node
+//.
+//. Common Node API's wrapped to return [Fluture][] Futures.
+//.
+//. ## API
+
+import Future from 'fluture';
+
+//# once :: String -> EventEmitter -> Future Error a
+//.
+//. Resolve a Future with the first event emitted over
+//. the given event emitter under the given event name.
+//.
+//. When the Future is cancelled, it removes any trace of
+//. itself from the event emitter.
+//.
+//. ```js
+//. > const emitter = new EventEmitter ();
+//. > setTimeout (() => emitter.emit ('answer', 42), 100);
+//. > once ('answer') (emitter);
+//. Future.of (42);
+//. ```
+export const once = event => emitter => Future ((rej, res) => {
+  const removeListeners = () => {
+    emitter.removeListener ('error', onError);
+    emitter.removeListener (event, onEvent);
+  };
+  const onError = x => {
+    removeListeners ();
+    rej (x);
+  };
+  const onEvent = x => {
+    removeListeners ();
+    res (x);
+  };
+  emitter.once ('error', onError);
+  emitter.once (event, onEvent);
+  return removeListeners;
+});
+
+//# buffer :: ReadableStream a -> Future Error (Array a)
+//.
+//. Buffer all data on a Stream into a Future of an Array.
+//.
+//. When the Future is cancelled, it removes any trace of
+//. itself from the Stream.
+//.
+//. ```js
+//. > const stream = new Readable ();
+//. > setTimeout (() => stream.push ('hello').push ('world'), 100);
+//. > buffer (stream);
+//. Future.of ([Buffer.from ('hello'), Buffer.from ('world')]);
```

Comment:
Is this example meant to work being `Readable` = `require('stream').Readable`?

Reply:
Yes. Does it? :p

Reply:
```js
const stream = new Readable ({read: () => {}});
setTimeout (() => stream.push('hello') && stream.push('world') && stream.push(null), 100);
buffer (stream)
```

Reply:
`stream.push` returns a `Boolean` and throws if no `read` function was defined.